feat(config)!: determine active platform via API and make plugin generic for all platforms / propositions

NabDevs opened the pull request:
- Add dynamic proposition detection based on API key validation
- Implement minimal PDK boot system
- Add namespace migration service for proposition-based configuration
- Complete hook system implementation with all required methods

Resolves INT-1049

[Needs specific PDK version.](https://github.com/myparcelnl/pdk/pull/379)

FreekVR committed:
build: remove platform-specific builds

removes platform-specific builds (distributed zips) per platform as the platform will be fetched in runtime in the plugin itself

fixes INT-1051

NabDevs committed:
feat: implement API key-based proposition detection

- Add dynamic proposition detection based on API key validation
- Implement minimal PDK boot system
- Add namespace migration service for proposition-based configuration
- Complete hook system implementation with all required methods

Resolves API key proposition detection requirements

NabDevs committed:
feat: implement API key-based proposition detection

- Add dynamic proposition detection based on API key validation
- Implement minimal PDK boot system
- Add namespace migration service for proposition-based configuration
- Complete hook system implementation with all required methods

Resolves API key proposition detection requirements

FreekVR commented on controllers/MyParcelNLAdminSettingsController.php:
```diff
@@ -6,6 +6,6 @@
 
 /**
- * @ModuleActivated(moduleName='myparcelnl', redirectRoute='myparcelnl_settings')
+ * @ModuleActivated(moduleName='myparcelnl', redirectRoute='myparcelcom_settings')
```
I think this suggestion is correct?

joerivanveen replied:
This is a bigger problem I’m working on currently.

joerivanveen replied:
We decided to leave the name ‘myparcelnl’ to avoid enormous upgrade / migration problems and I have introduced a constant to that effect `MyParcelNL::MODULE_NAME`

FreekVR commented on src/Service/NamespaceMigrationService.php:
```diff
@@ -0,0 +1,192 @@
+    /**
+     * Migrate any embedded namespaces within JSON values
+     */
+    private function migrateJsonNamespaces($value): string
```
this also seems to be correct, in fact there's an early return of the original value if its not a `string`, which would cause an error in this case

joerivanveen replied:
Yes implemented it.

joerivanveen committed:
fix: implement feedback

joerivanveen committed:
fix: implement feedback

joerivanveen committed:
refactor: remove namespace migration

FreekVR committed:
fix: remove call to removed `getPropositionName`

FreekVR committed:
fix: compatibility fixes for updated proposition implementation

FreekVR committed:
fix: use legacy carrier names (as saved in DB) for country mapping

FreekVR committed:
fix(carrier): add missing country/carrier mapping for myparcel-italie

FreekVR committed:
fix(brt): fix issue where domestic shipment label could not be printed due to missing state

FreekVR committed:
chore: bump pdk packages

FreekVR committed:
chore: remove upgrade commands removed in pdk app-builder v2

FreekVR committed:
feat: compile zip in semantic-release, not through pdk-builder

FreekVR committed:
build: bump @myparcel-pdk/app-builder to v4, removes zip commands

bump to v4 rather than a new v2 version due to issues with the versioning

FreekVR committed:
fix: fix the build folder name

FreekVR committed:
build(deps): bump myparcelnl/pdk to v3 and update all locks

FreekVR committed:
fix(sdk): fix missing explicit SDK dependency and update namespace for SDK v10

FreekVR committed:
fix: fix missing or empty classes

FreekVR committed:
fix(proposition): fix several missing new-to-legacy carrier name mappings

FreekVR committed:
test: fix numerous test issues due to proposition upgrade of the pdk (v3)

FreekVR committed:
fix(pdk): distribute weight when multiple shipments

fixes INT-1343

FreekVR committed:
build(deps): remove unused @vueuse/core dependency

FreekVR committed:
ci: remove deprecated "reviewers" property in dependabot config

FreekVR committed:
refactor: rename `PdkContextService` to `PsPdkContextService` and remove unused functions

FreekVR committed:
fix(admin): fix myparcel settings not rendering

fix myparcel settings not rendering due to routes not being registered with a correct namespace

FreekVR commented on release.config.cjs:
```diff
@@ -27,18 +27,14 @@ module.exports = {
     addChangelogPlugin(),
     addExecPlugin({
-      prepareCmd: `yarn pdk-builder release --root-command "${process.env.PDK_ROOT_COMMAND}" --version $\{nextRelease.version} -vvv`,
+      prepareCmd: `yarn pdk-builder release --root-command "${process.env.PDK_ROOT_COMMAND}" --version $\{nextRelease.version} -vvv && zip -r ./dist/myparcel-woocommerce-$\{nextRelease.version}.zip -C dist .`,
```
-C is the preserve case option and dist is simply the positional argument to zip that directiory

FreekVR committed:
build: fix plugin build zip being named "myparcel-woocommerce"

FreekVR committed:
test: add missing hooks to test assertions

FreekVR committed:
fix: do not run namespace migration in this plugin version

see !396 instead

FreekVR committed:
build(deps): update pdk to v3.1.1

FreekVR committed:
fix(upgrade): replace + in upgrade filenames (used for PR builds)
